modules/platforms/cpp/binary/src/impl/binary/binary_reader_impl.cpp

```diff
@@ -777,6 +777,24 @@ namespace ignite
                 return ReadCollectionSizeUnprotected();
             }
 
+            bool BinaryReaderImpl::IsNull(const char* fieldName)
+            {
+                CheckRawMode(false);
+                CheckSingleMode(true);
+
+                InteropStreamPositionGuard<InteropInputStream> positionGuard(*stream);
+
+                int32_t fieldId = idRslvr->GetFieldId(typeId, fieldName);
+                int32_t fieldPos = FindField(fieldId);
+
+                if (fieldPos <= 0)
+                    return true;
+
+                stream->Position(fieldPos);
+
+                return SkipIfNull();
```

Comment:
`SkipIfNull` will advance the position, is that ok here?

Reply:
Yeah, there is a position guard in the beginning. It makes sure the position is going to be restored when the function is left.